### PR TITLE
Added Reimage API support for ephemeral os disks in single VM

### PR DESCRIPTION
**Pull request by hari-bodicherla**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
There are model validation failures for this new operation that need to be fixed, please see the [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/443941076) for details.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json [hunk 1]
```diff
@@ -3202,13 +3256,13 @@
             "description": "The name of the VM scale set."
           },
           {
-            "name": "vmInstanceIDs",
+            "name": "vmScaleSetReimageInput",
             "in": "body",
             "required": false,
             "schema": {
-              "$ref": "#/definitions/VirtualMachineScaleSetVMInstanceIDs"
+              "$ref": "#/definitions/VirtualMachineScaleSetReimageParameters"
```
[the remark this answers is not in the document]
Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json]: hi, VirtualMachineScaleSetReimageParameters is the wrapper class around the VirtualMachineScaleSetVMInstanceIDs containing new inputs like osdisk, datadisk list of lun values and temp disk reiamge inputs for the customer. This needs to be added

**Comment by jhendrixMSFT** [comment]
If you do a search in the log for the example name, e.g. `Reimage a Virtual Machine.`, you will find the relevant failures.  Here's an example.
`
 error : 
operationId: VirtualMachines_Reimage
scenario: Reimage a Virtual Machine.
source: request
responseCode: ALL
severity: 0
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - properties
      - location
  message: 'Additional properties not allowed: properties,location'
  path: ''
  title: /definitions/VirtualMachineReimageParameters
  description: Paramaters for Reimaging Virtual Machine.
  position:
    line: 4917
    column: 40
  url: >-
    specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/compute.json
`

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/examples/ReimageVirtualMachine.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "myResourceGroup",
+    "api-version": "2018-06-01",
+    "vmName": "myVMName",
+    "parameters": {
+      "location": "West US",
```
Comment (jhendrixMSFT) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2018-06-01/examples/ReimageVirtualMachine.json]: This isn't in the schema and is causing one of the model validation failures.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "myResourceGroup",
+    "api-version": "2018-10-01",
+    "vmName": "myVMName",
+    "parameters": {
+      "location": "West US",
```
Comment (jhendrixMSFT) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json]: This isn't in the schema and is causing one of the model validation failures.

Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json]: Joel, is location parameter culprit?

Comment (jhendrixMSFT) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json]: Yes I believe so.  Note that you can run this validation locally to verify the fix before updating the PR, see the docs [here](https://github.com/azure/oav).

Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json]: Thanks Joel for the tool to validate locally. I had updated the examples and validated it.

Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/examples/ReimageVirtualMachine.json]: However I see there are other validation errors coming for others unrelated to mine

**Comment by jhendrixMSFT** [comment]
@hari-bodicherla while it would be nice to clean up the other failures, since they weren't introduced in this PR it's not necessary.  Let me know how you want to proceed.

**Comment by hari-bodicherla** [comment]
Hi Joel, I had cleaned up the other failures too which are unrelated to my changes and sent new iteration . Thanks..

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**